atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/keyvalue/api/cache/LockWatchValueScopingCacheImpl.java

```diff
@@ -192,10 +195,11 @@ private synchronized void updateStores(TransactionsLockWatchUpdate updateForTran
         Multimap<Sequence, StartTimestamp> reversedMap = createSequenceTimestampMultimap(updateForTransactions);
 
         // Without this block, updates with no events would not store a snapshot.
-        currentVersion.map(LockWatchVersion::version).map(Sequence::of).ifPresent(sequence -> Optional.ofNullable(
-                        reversedMap.get(sequence))
-                .ifPresent(startTimestamps ->
-                        snapshotStore.storeSnapshot(sequence, startTimestamps, valueStore.getSnapshot())));
+        currentVersion
+                .map(LockWatchVersion::version)
+                .map(Sequence::of)
+                .ifPresent(sequence ->
+                        snapshotStore.storeSnapshot(sequence, reversedMap.get(sequence), valueStore.getSnapshot()));
```

Comment:
multimaps don't return nulls, they return empty collections, so removed the optional wrapping